### ISSUE TITLE
Create new page in existing incognito window

### ISSUE THREAD

**Issue by ankur-lt** [GoogleChrome/lighthouse#15153]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://google.com

### What happened?

I launched chromium with the following command:
```sh
/Applications/Chromium.app/Contents/MacOS/Chromium --remote-debugging-port=9223 --incognito
```
Chromium version: Version 115.0.5761.0 (Developer Build) (x86_64)

Then I opened the URL via the Node.js script as follows:
```js
const lighthouse = require('lighthouse');

(async () => {
    let options, flags;
    console.log('Opening URL via lighthouse')

    options = {
        output: 'html',
        onlyCategories: ['performance', 'accessibility', 'best-practices', 'seo', 'pwa'],
        port: 9223,
    };

    const url = "https://google.com"

    await lighthouse(url, options, flags);
})()

```

The URL opens up in the existing chromium incognito window.

When I execute the same commands via the CLI
```sh
lighthouse https://google.com --port=9223
```
This opens up a new window which is not incognito.

### What did you expect?

The behaviour should be similar via CLI or the JS script as we are using the similar params and port number. The URL should be opened in the incognito window that is already open while opening the URL via CLI. 
I am running a Playwright test and it opens the browser in incognito mode by default. After the test runs, I want to trigger this CLI command to get the lighthouse report in incognito itself so that the cookies and storage for the domain is still present in the browser.

### What have you tried?

I have also tried explicitly passing the CLI arg `--chrome-flags="--incognito"`,  but it still opened in normal window.

### How were you running Lighthouse?

CLI, node

### Lighthouse Version

v9.6.8

### Chrome Version

115

### Node Version

18

### OS

MacOS Ventura

**Comment by ankur-lt** [comment]
@connorjclark , I already checked it, and it works as expected(https://github.com/GoogleChrome/lighthouse/blob/main/docs/puppeteer.md). 
As per I mentioned, I want to use the lighthouse CLI to open the lighthouse URL in the existing incognito window. I am using Playwright's client server architecture where the test is executed on a remote machine.

Ideally, any scenario that the lighthouse Node.js library is performing should ideally be reproduced by the CLI too, but here its not happening. Can you help me with the equivalent CLI command to be used for this?

**Comment by connorjclark** [comment]
I do not see that behavior when running from a node script. It still opens in a non-incognito window, not the existing 
incognito one.

This is where we connect to the browser using puppeteer: https://github.com/GoogleChrome/lighthouse/blob/333a6674613dd739382be0525f6a399cc0a2dc65/core/gather/navigation-runner.js#L356

Calling `lhBrowser.browserContexts()` returns just the single browser context puppeteer just created when connecting to the browser debug port, by definition:

<img width="772" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/4071474/52e5f56e-d56e-4e34-a993-f17a913d61c9">

(more are added here as pages calls window.open, or the node script calls `createIncognitoBrowserContext`)

It's not immediately clear if it is possible to create a browser context pointing to an existing browser window in puppeteer. If it were, we could just replace `lhBrowser.newPage()` with `thatContext.newPage()`.

Must your test be running in incognito? If the test runs in a fresh profile, it should be equivalent.

**Comment by ankur-lt** [comment]
You can try the example provided in my first comment for opening the lighthouse URL in the existing incognito window.
Just open a browser via the CLI command
```
/Applications/Chromium.app/Contents/MacOS/Chromium --remote-debugging-port=9223 --incognito
```
The run the following Node.js script:
```js
const lighthouse = require('lighthouse');

(async () => {
    let options, flags;
    console.log('Opening URL via lighthouse')

    options = {
        output: 'html',
        onlyCategories: ['performance', 'accessibility', 'best-practices', 'seo', 'pwa'],
        port: 9223,
    };

    const url = "https://google.com"

    await lighthouse(url, options);
})()
```

This opens the webpage in the existing  incognito window.

I am trying to achieve the same using the lighthouse CLI command below:
```sh
lighthouse https://google.com --port=9223
```
But this is opening the webpage in a non-incognito mode. I want to open it in the existing incognito window for my use case. Ideally, as the params provided are the same, the CLI command should open the webpage in the existing incognito window itself.

**Comment by ankur-lt** [comment]
Updated the issue text.
I am using lighthouse v9.6.8 and chromium 115. Using v9.6.8 opens the webpage in the existing incognito window.
@connorjclark

**Comment by ankur-lt** [comment]
Looks like this is an issue with the lighthouse  v10. The CLI command with lighthouse `v9.8.6` is also opening the webpage in the existing incognito window.
I had lighthouse installed globally with `v10.2.0`, hence I was getting the above issue. 

Is this a bug with v10? @connorjclark

**Comment by ankur-lt** [comment]
@connorjclark , can you confirm if this is a bug with v10 and if there's any workaround for this?

**Comment by adamraine** [comment]
I can confirm the different behavior between v9 and v10. The `--legacy-navigation` flag *did not* affect anything for me in v10 though.

**Comment by adamraine** [comment]
So I think this is the change which caused this https://github.com/GoogleChrome/lighthouse/pull/14012.

Lighthouse v9 will attempt to create a new page using a `GET` request which throws an error in modern versions of chrome and then falls into the fallback case where we use an existing page.

Lighthouse v10 with `--legacy-navigation` performs a `PUT` request which does not throw an error and successfully creates a new page.

Basically, creating a new page is the expected behavior and using an existing page is the fallback behavior. It just so happens that Lighthouse v9 will always hit the fallback case due to an incompatibility with modern Chrome versions.

**Comment by adamraine** [comment]
@ankur-lt if you want to reuse an existing incognito page you can use puppeteer with Lighthouse:

```js
import lighthouse from 'lighthouse';
import puppeteer from 'puppeteer';

const browser = await puppeteer.connect({
  browserURL: 'http://localhost:9222',
})
const page = (await browser.pages())[0];
await lighthouse('https://example.com', undefined, undefined, page);

browser.disconnect();
```

I'm not sure if we can use CDP in the latest Chrome to create a new page in an existing incognito window. Might be worth investigating if it's possible.

**Comment by ankur-lt** [comment]
Thanks @adamraine for checking this. My use case here relates to opening a new tab in the existing incognito window or opening a headless incognito window to get the lighthouse report as the test runs by default in incognito in Playwright.

I had tried using lighthouse v9, but as per you confirmed above, it uses the existing page of the browser in latest versions rather than creating a new tab. This hampers the test execution which is taking place in the initial tab. Thus I cannot use v9.

I tried with the `PUT /json/new` request when the browser is opened in an incognito window. It opened the regular Chrome tab, not in incognito.
Do we have any workaround for this, or we need to create a bug with Chromium?

**Comment by adamraine** [comment]
> Do we have any workaround for this, or we need to create a bug with Chromium?

I don't think there is a workaround, I will need to ask internally if this is a limitation of the CDP protocol.

**Comment by adamraine** [comment]
I spoke with CDP folks internally, and this is currently not possible and probably will not ever be supported because it is a high-level concept specific to only some browser clients (e.g. desktop but not android/ios)

Even if this *does* get support somehow there isn't anything for us to do here, since we would want the user to pass in their desired puppeteer page instead of making it ourselves.
